crates/project-model/src/workspace.rs

```diff
@@ -815,20 +822,21 @@ fn project_json_to_crate_graph(
 }
 
 fn cargo_to_crate_graph(
+    crate_graph: &mut CrateGraph,
+    proc_macros: &mut ProcMacroPaths,
     load: &mut dyn FnMut(&AbsPath) -> Option<FileId>,
     rustc: Option<&(CargoWorkspace, WorkspaceBuildScripts)>,
     cargo: &CargoWorkspace,
     sysroot: Option<&Sysroot>,
     rustc_cfg: Vec<CfgFlag>,
     override_cfg: &CfgOverrides,
+    // Don't compute cfg and use this if present
+    forced_cfg: Option<CfgOptions>,
     build_scripts: &WorkspaceBuildScripts,
     target_layout: TargetLayoutLoadResult,
     channel: Option<ReleaseChannel>,
-) -> (CrateGraph, ProcMacroPaths) {
```

Comment:
Can we keep this as return values and instead of merge the graphs in the sysroot part? At worst we'll have some dependency duplicates for the sysroot right now, but I'd like this change to be as non-invasive as possible (and in theory we would have a crate dedup step at some point anyways https://github.com/rust-lang/rust-analyzer/issues/14476).

Reply:
👍 
How it can cause duplicate crates? I thought we add sysroot for a cargo project only one time. And the current way is not much different from a merge, I think it would create duplicates as well.

Reply:
Oh I assume this was because of possibly duplicates or something, then I don't see why we need this change at all. We should just `CrateGraph::extend` the graph in this case afterwards without changing the function signature here